Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/TransformComponent.cpp

```diff
@@ -954,6 +955,13 @@ namespace AzToolsFramework
                 return AZ::Failure(AZStd::string("You cannot set an entity to be a child of a closed container!"));
             }
 
+            // Don't allow entities to be parented outside their container.
+            AzToolsFramework::EntityIdList entities = m_focusModeInterface->GetFocusedEntities(GetEntityContextId());
+            if (std::find(entities.begin(), entities.end(), actualValue) == entities.end())
+            {
+                return AZ::Failure(AZStd::string("You cannot set an entity to be a child of a different container!"));
```

Comment:
nit : 'container is a bit too generic'. You can instead say something like "You can only set a parent as one of the entities belonging to the focused prefab"

Reply:
Change made.